statev2/state/src/replication/log_store.rs

```diff
@@ -195,18 +224,12 @@ impl Storage for LogStore {
     fn initial_state(&self) -> RaftResult<RaftState> {
         // Read the hard state
         let tx = self.db.new_read_tx().map_err(RaftError::from)?;
-        let hard_state: ProtoStorageWrapper<HardState> = tx
-            .read(RAFT_METADATA_TABLE, &HARD_STATE_KEY.to_string())
-            .map_err(RaftError::from)?
-            .unwrap_or_default();
-        let conf_state: ProtoStorageWrapper<ConfState> = tx
-            .read(RAFT_METADATA_TABLE, &CONF_STATE_KEY.to_string())
-            .map_err(RaftError::from)?
-            .unwrap_or_default();
+        let hard_state = self.read_hard_state_with_tx(&tx)?;
```

Comment:
looks like this no longer uses a default `conf_state` or `hard_state` if it's not already written to the DB, should we ensure that a default is written before this method gets called?